app/charts/shared/brush/index.tsx

```diff
@@ -79,33 +83,63 @@ export const BrushTime = () => {
     brushHandleFillColor,
     labelFontSize,
   } = useChartTheme();
-  const { chartType, bounds, xScaleTimeRange, xDimension } =
+  const { bounds, xScaleTimeRange, xDimension } =
     useChartState() as ChartWithInteractiveXTimeRangeState;
+  const { chartType } = useChartState() as
+    | ChartWithInteractiveYTimeRangeState
+    | ChartWithInteractiveXTimeRangeState;
+  const { yScaleTimeRange, yDimension } =
+    useChartState() as ChartWithInteractiveYTimeRangeState;
   const formatDateUnit = useTimeFormatUnit();
   const formatDate = (date: Date) => {
+    if (chartType === "bar") {
+      return formatDateUnit(
+        date,
+        (yDimension as TemporalDimension | TemporalEntityDimension).timeUnit
+      );
+    }
     return formatDateUnit(
       date,
       (xDimension as TemporalDimension | TemporalEntityDimension).timeUnit
     );
   };
   const { getX } = useChartState() as LinesState | AreasState;
   const { getXAsDate, allData } = useChartState() as ColumnsState;
-  const getDate = chartType === "column" ? getXAsDate : getX;
+  const { getYAsDate } = useChartState() as BarsState;
+  const getDate = (() => {
+    if (chartType === "bar") {
+      return getYAsDate;
+    }
+    if (chartType === "column") {
+      return getXAsDate;
+    }
+    return getX;
+  })();
   const fullData = allData;
 
   // Brush dimensions
   const { width, margins, chartHeight } = bounds;
   const brushLabelsWidth =
-    getTextWidth(formatDate(xScaleTimeRange.domain()[0]), {
-      fontSize: labelFontSize,
-    }) +
-    getTextWidth(" - ", { fontSize: labelFontSize }) +
-    getTextWidth(formatDate(xScaleTimeRange.domain()[1]), {
-      fontSize: labelFontSize,
-    }) +
-    HANDLE_HEIGHT;
+    chartType === "bar"
+      ? getTextWidth(formatDate(yScaleTimeRange.domain()[0]), {
+          fontSize: labelFontSize,
+        }) +
+        getTextWidth(" - ", { fontSize: labelFontSize }) +
+        getTextWidth(formatDate(yScaleTimeRange.domain()[1]), {
+          fontSize: labelFontSize,
+        }) +
+        HANDLE_HEIGHT
+      : getTextWidth(formatDate(xScaleTimeRange.domain()[0]), {
+          fontSize: labelFontSize,
+        }) +
+        getTextWidth(" - ", { fontSize: labelFontSize }) +
+        getTextWidth(formatDate(xScaleTimeRange.domain()[1]), {
+          fontSize: labelFontSize,
+        }) +
+        HANDLE_HEIGHT;
   const brushWidth = width - brushLabelsWidth - margins.right;
-  const brushWidthScale = xScaleTimeRange.copy();
+  const brushWidthScale =
+    chartType === "bar" ? yScaleTimeRange.copy() : xScaleTimeRange.copy();
```

Comment:
Nit: @squiles I wonder if we can't extract something like

```typescript
const scaleTimeRange = chartType === "bar" ? yScaleTimeRnage : xScaleTimeRange;
```

and avoid having ternaries and duplicating the logic – I think it's all the same except for scale variable 🤔